File: MediaLoader/src/main/java/com/app/medialoader/data/DefaultDataSourceFactory.java
package com.app.medialoader.data;

import com.app.medialoader.MediaLoaderConfig;
import com.app.medialoader.data.file.FileDataSource;
import com.app.medialoader.data.file.RandomAcessFileDataSource;
import com.app.medialoader.data.file.cleanup.DiskLruCache;
import com.app.medialoader.data.file.cleanup.SimpleDiskLruCache;
import com.app.medialoader.data.url.DefaultUrlDataSource;
import com.app.medialoader.data.url.UrlDataSource;

import java.io.File;

/**
 * 数据源默认生产工厂
 *
 * @author vincanyang
 */
public final class DefaultDataSourceFactory {

    public static UrlDataSource createUrlDataSource(String url) {
        return new DefaultUrlDataSource(url);
    }

    public static UrlDataSource createUrlDataSource(DefaultUrlDataSource dataSource) {
        return new DefaultUrlDataSource(dataSource);
    }

    public static FileDataSource createFileDataSource(File file, DiskLruCache diskLruStorage) {
        return new RandomAcessFileDataSource(file, diskLruStorage);
    }

    public static DiskLruCache createDiskLruCache(MediaLoaderConfig mediaLoaderConfig) {
        return new SimpleDiskLruCache(mediaLoaderConfig);
    }
}

File: MediaLoader/src/main/java/com/app/medialoader/data/DefaultDataSourceFactory.kt
package com.app.medialoader.data

import com.app.medialoader.MediaLoaderConfig
import com.app.medialoader.data.file.FileDataSource
import com.app.medialoader.data.file.RandomAcessFileDataSource
import com.app.medialoader.data.file.cleanup.DiskLruCache
import com.app.medialoader.data.file.cleanup.SimpleDiskLruCache
import com.app.medialoader.data.url.DefaultUrlDataSource
import com.app.medialoader.data.url.UrlDataSource
import java.io.File

object DefaultDataSourceFactory {

    fun createUrlDataSource(url: String?): UrlDataSource {
        return DefaultUrlDataSource(url)
    }

    fun createUrlDataSource(dataSource: DefaultUrlDataSource): UrlDataSource {
        return DefaultUrlDataSource(dataSource)
    }

    fun createFileDataSource(file: File, diskLruStorage: DiskLruCache?): FileDataSource {
        return RandomAcessFileDataSource(file, diskLruStorage)
    }

    fun createDiskLruCache(mediaLoaderConfig: MediaLoaderConfig): DiskLruCache {
        return SimpleDiskLruCache(mediaLoaderConfig)
    }
}